staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -544,30 +608,53 @@ func (e *Store) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 				return nil, nil, apierrors.NewConflict(qualifiedResource, name, fmt.Errorf(OptimisticLockErrorMsg))
 			}
 		}
-		if err := rest.BeforeUpdate(e.UpdateStrategy, ctx, obj, existing); err != nil {
-			return nil, nil, err
-		}
-		// at this point we have a fully formed object.  It is time to call the validators that the apiserver
-		// handling chain wants to enforce.
-		if updateValidation != nil {
-			if err := updateValidation(ctx, obj.DeepCopyObject(), existing.DeepCopyObject()); err != nil {
+		var cleanup func(runtime.Object, runtime.Object, error)
+
+		if e.BeginUpdate != nil {
+			fn, err := e.BeginUpdate(obj, existing, options)
```

Comment:
likewise, this could call BeginUpdate multiple times. Is that OK?

Reply:
Note that in an extreme case, the same request could come through as a create, then an update, then a create, then an update, and get retried each time.

Reply:
I think it has to be.  It's not ideal, but all the logic to fetch `obj` and `existing` are buried inside that loop.  I don't think it's worth making that more complicated when we can just document it as part of the contract

Reply:
Agree with Tim - I think it's - assuming we document the contract it sounds fine to me.

Reply:
That sounds right. This should have the same lifecycle as PrepareForUpdate/BeforeUpdate/admission, which happens inside the GuaranteedUpdate loop and can be invoked multiple times (on a fresh copy of the object) for a single REST request.